internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/WriteComputeWithActiveActiveReplicationTest.java

```diff
@@ -0,0 +1,355 @@
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key2, val2T1);
+
+    // val2T3 update should be applied since its timestamp t3 is greater than the existing timestamp t2 for age field
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key2, val2T3);
+    validatePersonV1Record(storeName, dc0RouterUrl, key2, "val2f1_b", 40);
+    validatePersonV1Record(storeName, dc1RouterUrl, key2, "val2f1_b", 40);
+
+    // val2T2Prime update should be applied since its timestamp t2 is equal to the existing timestamp t2 for name field
+    // TODO: is this expected behavior though? It seems this can cause divergence?
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(0)), storeName, key2, val2T2Prime);
+    validatePersonV1Record(storeName, dc0RouterUrl, key2, "val2f1_b_prime", 40);
+    validatePersonV1Record(storeName, dc1RouterUrl, key2, "val2f1_b_prime", 40);
+
+    /*
+    * Case: Add a new field and remove an existing field in the new schema.
+    * The final record should contain both removed and added fields.
+    */
+    String key3 = "key3";
+    GenericRecord val3 = new GenericData.Record(valueSchemaV1);
+    val3.put(PERSON_F1_NAME, "val3f1");
+    val3.put(PERSON_F2_NAME, 30);
+
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key3, val3);
+    validatePersonV1Record(storeName, dc0RouterUrl, key3, "val3f1", 30);
+    validatePersonV1Record(storeName, dc1RouterUrl, key3, "val3f1", 30);
+
+    assertCommand(parentControllerClient.addValueSchema(storeName, valueSchemaV2.toString()));
+
+    UpdateBuilder ubKv3 = new UpdateBuilderImpl(wcSchemaV2);
+    ubKv3.setNewFieldValue(PERSON_F3_NAME, "val3f3");
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(0)), storeName, key3, ubKv3.build());
+    validatePersonV1V2SupersetRecord(storeName, dc0RouterUrl, key3, "val3f1", 30, "val3f3");
+    validatePersonV1V2SupersetRecord(storeName, dc1RouterUrl, key3, "val3f1", 30, "val3f3");
+
+    // Removed field, when not set should, have a default value
+    String key3Prime = "key3Prime";
+    GenericRecord val3Prime = new GenericData.Record(valueSchemaV2);
+    val3Prime.put(PERSON_F1_NAME, "val3PrimeF1");
+    val3Prime.put(PERSON_F3_NAME, "val3PrimeF3");
+    sendStreamingRecord(dcSystemProducer.get(childDatacenters.get(1)), storeName, key3Prime, val3Prime);
+    validatePersonV1V2SupersetRecord(storeName, dc0RouterUrl, key3Prime, "val3PrimeF1", -1, "val3PrimeF3");
+    validatePersonV1V2SupersetRecord(storeName, dc1RouterUrl, key3Prime, "val3PrimeF1", -1, "val3PrimeF3");
+  }
+
+  private void validatePersonV1V2SupersetRecord(
+      String storeName,
+      String routerUrl,
+      String key,
+      String expectedField1,
+      int expectedField2,
+      String expectedField3) {
+    try (AvroGenericStoreClient<Object, Object> client = ClientFactory
```

Comment:
nit: might be more efficient if we can create just one thin client and reuse it in all validation. As I believe warmup takes a bit of time.

Reply:
Sure